### ISSUE TITLE
zfs_vn_rdwr panic

### ISSUE THREAD

**Issue by mattmacy** [zfsonfreebsd/ZoF#14]
~30 minutes in to ZTS:
```

panic: VERIFY(ioflag == 0) failed

cpuid = 15
time = 1547608919
KDB: stack backtrace:
db_trace_self_wrapper() at db_trace_self_wrapper+0x2b/frame 0xfffffe00a69d7030
vpanic() at vpanic+0x1b4/frame 0xfffffe00a69d7090
spl_panic() at spl_panic+0x4f/frame 0xfffffe00a69d7110
zfs_vn_rdwr() at zfs_vn_rdwr+0xbb/frame 0xfffffe00a69d71e0
dump_bytes_cb() at dump_bytes_cb+0x148/frame 0xfffffe00a69d7270
taskq_run() at taskq_run+0x34/frame 0xfffffe00a69d72a0
taskqueue_run_locked() at taskqueue_run_locked+0x10c/frame 0xfffffe00a69d7300
taskqueue_thread_loop() at taskqueue_thread_loop+0x88/frame 0xfffffe00a69d7330
fork_exit() at fork_exit+0x84/frame 0xfffffe00a69d7370
fork_trampoline() at fork_trampoline+0xe/frame 0xfffffe00a69d7370
--- trap 0, rip = 0, rsp = 0, rbp = 0 ---
KDB: enter: panic
[ thread pid 0 tid 101558 ]
Stopped at      kdb_enter+0x3b: movq    $0,kdb_why
```

**Comment by kmoore134** [comment]
![crash1](https://user-images.githubusercontent.com/1933088/51267673-47aac300-198c-11e9-8bb3-62fd4eb5c2e2.png)
![crash2](https://user-images.githubusercontent.com/1933088/51267674-49748680-198c-11e9-933c-be37c02c2563.png)

Looks like same issue in the delegate tests we trigger same panic.

**Comment by mattmacy** [comment]
fixed by 1f9c6ced88a469e91d35f6ff0b7d0cb420abf574
